tools/prune-old-builds.py

```diff
@@ -0,0 +1,81 @@
+import os
+import shutil
+import argparse
+import subprocess
+from pathlib import Path
+from datetime import datetime
+
+
+FOURTEEN_DAYS = 14
+
+
+def how_old(date: datetime) -> int:
+    """Return a integer that represents how many days has passed given a date"""
+    return (datetime.now() - date).days
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser("Prune old builds")
+    parser.add_argument(
+        "builds_path",
+        help="The path where the builds are stored.",
+        nargs=1,
+        type=str)
+    parser.add_argument(
+        "repository_url",
+        help="The url to the repository.",
+        nargs=1,
+        type=str)
+    args, _ = parser.parse_known_args()
+
+    builds_path = Path(args.builds_path[0])
+    # https://github.com/quick-lint/quick-lint-js.git
+    repository_url = args.repository_url[0]
+    repo_name = 'quick-lint-js'
+
+    if not os.path.exists(builds_path):
+        print(
+            f"error: The directory \033[1m{builds_path}\033[0m does not exists.")
+        exit(1)
+
+    if not os.path.exists(repo_name):
+        print("Cloning repository")
+        subprocess.run(['git', 'clone', '-q', repository_url])
+
+    print('Getting all commits of the repository\n')
+    commits = subprocess.getoutput(
```

Comment:
`subprocess.getoutput` has a few problems:

* It mixes error output with normal output into the returned string. This means that network errors won't be visible to the human at all.
* **Must fix**: If the command fails, the script proceeds as if nothing bad happens. This is very bad. If a network error occurs, then `git rev-list --all --remotes` won't run. If `git rev-list --all --remotes` doesn't run, no commits end up in the output. If no commits end up in the output, we'll attempt to delete our archive of old builds!
* `subprocess.getoutput` always invokes a shell. This makes it harder to safely inject arguments (such as `repo_name` or `repository_url`.

Use `subprocess.check_output` with a list instead, which avoids all these problems. https://docs.python.org/3/library/subprocess.html?highlight=subprocess#subprocess.check_output